samples/msal-node-samples/client-credentials-with-cert-from-key-vault/test/client-credentials-with-cert-from-key-vault-aad.spec.ts

```diff
@@ -0,0 +1,100 @@
+import {
+    RETRY_TIMES,
+    validateCacheLocation,
+    NodeCacheTestUtils,
+} from "e2e-test-utils";
+import { ConfidentialClientApplication } from "@azure/msal-node";
+import config from "../config/AAD.json";
+import { getKeyVaultSecretClient } from "e2e-test-utils/src/KeyVaultUtils";
+import { getCertificateInfo } from "e2e-test-utils/src/CertificateUtils";
+import {
+    ENV_VARIABLES,
+    LAB_CERT_NAME,
+    LAB_KEY_VAULT_URL,
+} from "e2e-test-utils/src/Constants";
+
+const TEST_CACHE_LOCATION = `${__dirname}/data/aad.cache.json`;
+
+const getClientCredentialsToken = require("../index");
+
+const cachePlugin = require("../../cachePlugin.js")(TEST_CACHE_LOCATION);
+
+const clientCredentialRequestScopes = ["https://graph.microsoft.com/.default"];
+
+describe("Client Credentials AAD Prod Tests", () => {
+    jest.retryTimes(RETRY_TIMES);
+    jest.setTimeout(90000);
+
+    let thumbprint: string;
+    let privateKey: string;
+    let x5c: string;
+    beforeAll(async () => {
+        await validateCacheLocation(TEST_CACHE_LOCATION);
```

Comment:
FYI: Im not sure how this works on JS but in MSAL.Net we always try to clear all the static caches we can before every test to ensure there are no weird collisions